src/main/java/org/noureddine/joularjx/cpu/PowermetricsMacOS.java

```diff
@@ -61,19 +62,17 @@ public double getCurrentPower(double cpuLoad) {
                 }
 
                 // looking for line fitting the: "<name> Power: xxx mW" pattern and add all of the associated values together
-                final var powerIndicatorIndex = line.indexOf(POWER_INDICATOR);
-
-                // we need an exit condition to avoid looping forever (since there are always new lines, the process being periodical)
-                // if we started processing power lines and we don't find any anymore, we've reached the end of this "page" so exit the loop
-                if(processingPower && powerIndicatorIndex < 0) {
-                    break;
-                }
+                // or, for Intel chips, the: "Intel energy model derived package power (CPUs+GT+SA): xxx W" pattern
+                final var powerIndicatorIndexM = line.indexOf(POWER_INDICATOR_M_CHIP);
```

Comment:
I would actually now avoid skipping all the headers line and check for the presence of the EFI line which should be indicative of an Intel CPU. This would then allow to avoid looking for both power indicators per line.

Reply:
I've created a readHeader() function that is called from the initialize. This detects the chip and is also only called once since the header is also only printed once.